module/VuFind/src/VuFind/RecordTab/CollectionList.php

```diff
@@ -117,6 +117,16 @@ public function getParams()
         return $this->getResults()->getParams();
     }
 
+    /**
+     * Is this tab active?
+     *
+     * @return bool
+     */
+    public function isActive()
+    {
+        return parent::isActive() && $this->getRecordDriver()->tryMethod('getHierarchyTrees');
```

Comment:
Seems like isCollection might be the safest bet of all:

```suggestion
        return parent::isActive() && $this->getRecordDriver()->tryMethod('isCollection');
```

Is there any reason not to do it this way?

Reply:
CollectionList can be active on subrecords and seems like not all of them implements is_hierarchy_id which is used in isCollection function check. That is why i was thinking about if using getHierarchyTopID would be better for this occasion :thinking:

Reply:
Is it possible this is just a configuration issue? The [hierarchy configuration](https://github.com/vufind-org/vufind/blob/dev/config/vufind/HierarchyDefault.ini#L2-L7) controls which records are considered to be collections and which are not, based on the Solr index. If you have records that act like collections but do not have is_hierarchy_id set, that sounds like something non-standard, and I wonder if it would make more sense to customize record drivers than to introduce non-standard behavior to the record tab.

Bottom line: my feeling is that, at least for the core code, this tab is meant to operate on collections, so the best check we can apply is confirming that the record under consideration is a collection. Anything else feels like introducing different rules here than everywhere else, which doesn't make sense to me. But as I say, it's been a while since I've reviewed all this, so maybe I'm overlooking something -- my feeling, though, is that if something is wrong, the problem runs deeper than just this tab, and maybe needs to be addressed somewhere else.

Reply:
Hmm, actually this comment clears up my thoughts. I think the isCollection check makes alot more sense now in this context :thinking:

Reply:
Great, thanks! I do think adding that check is an improvement, so if it works for you, I'll be happy to approve and merge this once the change is in place.